core/src/main/java/hudson/WebAppMain.java

```diff
@@ -189,7 +190,11 @@ public Locale get() {
 
             final FileAndDescription describedHomeDir = getHomeDir(event);
             home = describedHomeDir.file.getAbsoluteFile();
-            home.mkdirs();
+            try {
+                Files.createDirectories(home.toPath());
+            } catch (IOException e) {
+                throw new UncheckedIOException(e);
+            }
             LOGGER.info("Jenkins home directory: "+ home +" found at: " + describedHomeDir.description);
 
             // check that home exists (as mkdirs could have failed silently), otherwise throw a meaningful error
```

Comment:
At least adjust the comment, or delete the whole clause? Or rethrow `IOException` as `NoHomeDir`?